### PR TITLE
#162727472 Users should be able to like or dislike articles

### PR DESCRIPTION
**Pull request by daniellamarr**
#### What does this PR do?
- This PR allows a user to like or dislike articles on the platform
#### Description of Task to be completed?
- A user can like and unlike an article
- A user can dislike or un-dislike an article
- A user cannot like an article more than once
- A user cannot dislike an article more than once
- When an article has already been liked, a dislike automatically erases the like
- When an article has already been disliked, a like automatically erases the dislike
#### How should this be manually tested?
- Run `npm start`
- Access `/api/v1/arts/:artId/like` to like and unlike an article, where `artId` represents the id of the art to be liked
- Access `/api/v1/arts/:artId/dislike` to dislike and undislike an article, where `artId` represents the id of the art to be disliked
#### Any background context you want to provide?
- The user has to be signed up or logged in to access the authenticated routes. The user's token would be needed for these routes.
#### What are the relevant pivotal tracker stories?
[#162727472](https://www.pivotaltracker.com/story/show/162727472)
#### Screenshots (if appropriate)
screen shot 2019-01-24 at 1 41 57 pm 2
screen shot 2019-01-24 at 1 42 06 pm 2

#### Questions:
N/A

### CODE REVIEW

**Review by julietezekwe — APPROVED** [review]

**Comment by wombolo** [comment]
Kindly resolve merge conflict

File: src/controllers/ArtsController.js [hunk 1]
```diff
@@ -417,6 +419,235 @@ class ArtsController {
...
+          'Bad Request',
+          400,
+          'You have already liked this article',
+        );
+        return res.status(response.code).json(response);
+      }
+
+      LikeUnlike.like(artId);
+
+      const checkIfDisliked = await Like.findOne({
```
Comment (andela-dbamidele) [inline src/controllers/ArtsController.js:463]: What's the difference between this query and the other one? At what point will this code run?

File: src/controllers/ArtsController.js [hunk 2]
```diff
@@ -417,6 +419,235 @@ class ArtsController {
...
+
+  /**
+  * @static
+  * @desc POST /api/v1/arts/:artId/unlike
+  * @param {object} req
+  * @param {object} res
+  * @memberof ArtsController
+  * @returns {object} successful unlike
+  */
+  static async unlikeArticle(req, res) {
```
Comment (andela-dbamidele) [inline src/controllers/ArtsController.js]: I think you should be able to handle like and unlike in one controller.

File: src/controllers/ArtsController.js [hunk 3]
```diff
@@ -417,6 +419,235 @@ class ArtsController {
...
+
+  /**
+  * @static
+  * @desc POST /api/v1/arts/:artId/undislike
+  * @param {object} req
+  * @param {object} res
+  * @memberof ArtsController
+  * @returns {object} successful undislike
+  */
+  static async undislikeArticle(req, res) {
```
Comment (andela-dbamidele) [inline src/controllers/ArtsController.js]: Same goes for this.

File: src/controllers/UsersController.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ import { passwordHash, comparePassword } from '../helpers/passwordHash';
 import EmailNotificationAPI from '../helpers/EmailNotificationAPI';
 import basePath from '../helpers/basepath';
 
-const { User, Profile } = models;
+const { User, Profile, } = models;
```
Comment (andela-dbamidele) [inline src/controllers/UsersController.js]: Is this trailing comma necessary?

**Review by andela-dbamidele — CHANGES_REQUESTED** [review]

File: src/controllers/ArtsController.js (hunk 1 above)
Comment (daniellamarr) [reply src/controllers/ArtsController.js:463]: There are two functionalities implemented; `Like` and `Dislike`. Apparently, a user cannot dislike what he has already liked, neither can he like what he already disliked. So, in this case, I'm just checking if the user's details exist with specified art id on the Dislike model before completing the function.

At the point the user's details alongside the art id can be found in the Dislike model, I delete that record and complete the like for the user.

File: src/controllers/ArtsController.js (hunk 2 above)
Comment (daniellamarr) [reply src/controllers/ArtsController.js]: Yeah, I know it's possible. I thought that too but I felt separate controllers could work as well.

But if you suggest I have them in one controller, it's fine

File: src/controllers/ArtsController.js (hunk 3 above)
Comment (daniellamarr) [reply src/controllers/ArtsController.js]: Alright

File: src/controllers/UsersController.js (hunk 4 above)
Comment (daniellamarr) [reply src/controllers/UsersController.js]: Oh no, my bad.
Bad practice

File: src/controllers/ArtsController.js (hunk 1 above)
Comment (andela-dbamidele) [reply src/controllers/ArtsController.js:463]: I think I'll like to see a demo of this.

File: src/controllers/ArtsController.js (hunk 2 above)
Comment (andela-dbamidele) [reply src/controllers/ArtsController.js]: Yea. One controller.

**Review by wombolo — APPROVED** [review]
Looks Good

**Review by abejide001 — APPROVED** [review]
LGTM

**Pull request merged**